### PR TITLE
Add RBAC

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/1912

### CODE REVIEW

File: helm/pv-cleaner-operator-chart/templates/rbac.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+kind: ClusterRole
+metadata:
+  name: {{ .Values.clusterRoleName }}
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - persistentvolumes
+      - persistentvolumeclaims
+      - pods
```
Comment (tuommaki) [inline helm/pv-cleaner-operator-chart/templates/rbac.yaml]: Why does this need all verbs for PODs?

**Review by tuommaki — APPROVED** [review]
In general this looks good. One possibility for code simplification. I'm not too deep into the actual logic so I trust that this is properly tested. :slightly_smiling_face:

File: helm/pv-cleaner-operator-chart/templates/rbac.yaml (hunk 1 above)
Comment (corest) [reply helm/pv-cleaner-operator-chart/templates/rbac.yaml]: Hm, true, for pods it needs only `list` and `delete`.

File: service/controller/v1/resource/persistentvolume/update.go [hunk 2]
```diff
@@ -92,27 +102,30 @@ func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateState inter
 			return nil
 		}
 
-		if err := r.k8sClient.Batch().Jobs("kube-system").Delete(cleanupJob.Name, &metav1.DeleteOptions{}); err != nil {
+		err = r.k8sClient.Batch().Jobs("kube-system").Delete(cleanupJob.Name, &metav1.DeleteOptions{})
```
Comment (rossf7) [inline service/controller/v1/resource/persistentvolume/update.go]:
```suggestion
		err = r.k8sClient.Batch().Jobs(metav1.NamespaceSystem).Delete(cleanupJob.Name, &metav1.DeleteOptions{})
```

Minor but I like using the namespace constants for stuff like this.

**Review by rossf7 — APPROVED** [review]
Not deep into the logic but LGTM.

Nice to get RBAC enabled here.

**Pull request merged**